data/completion/complete.sh

```diff
@@ -100,17 +101,28 @@ _complete_from_snap() {
 
 }
 
-# _complete_from_snap_maybe calls _complete_from_snap if the command is in
-# bin/snap, and otherwise does bash-completion's _completion_loader (which is
-# what -D would've done before).
-_complete_from_snap_maybe() {
-    # catch /snap/bin and /var/lib/snapd/snap/bin
-    if [[ "$(which "$1")" =~ /snap/bin/ && ( -e /var/lib/snapd/snap/core/current/usr/lib/snapd/etelpmoc.sh || -e /snap/core/current/usr/lib/snapd/etelpmoc.sh ) ]]; then
-        _complete_from_snap "$1"
-        return $?
-    fi
-    # fallback to the old -D
-    _completion_loader "$1"
-}
+# this file can be sourced directly as e.g. /usr/lib/snapd/complete.sh, or via
+# a symlink from /usr/share/bash-completion/completions/. In the first case we
+# want to load the default loader; in the second, the specific one.
+if [[ "${BASH_SOURCE[0]}" =~ /complete\.sh$ ]]; then
```

Comment:
what happens if somebody has  a command called "complete.sh" for their snap? I think it's not valid app command but it's a valid alias, so something to consider